src/library_syscall.js

```diff
@@ -114,10 +114,12 @@ var SyscallsLibrary = {
       }
       return 0;
     },
-    doDup: function(path, flags, suggestFD) {
+    doDup: function(stream, suggestFD) {
       var suggest = FS.getStream(suggestFD);
       if (suggest) FS.close(suggest);
-      return FS.open(path, flags, 0, suggestFD, suggestFD).fd;
+
+      FS.streams[suggestFD] = stream;
```

Comment:
Ok, I read the libc docs more carefully now :) Yes, you are right. I'm surprised we didn't notice this before!

Docs for running tests are here: http://kripken.github.io/emscripten-site/docs/getting_started/test-suite.html

That test could be added in `tests/test_other.py`, there are some other filesystem tests there already.